lib/Conversion/PipelineToHW/PipelineToHW.cpp

```diff
@@ -115,15 +123,32 @@ class PipelineLowering {
       Type dataType = regIn.getType();
       Value dataReg;
       if (this->clockGateRegs) {
-        // Clock gate based on the valid signal.
-        dataReg = builder.create<seq::CompRegClockEnabledOp>(
-            stageTerminator->getLoc(), dataType, regIn, clock, dataValid,
-            regName, reset, /*resetValue*/ Value(), /*sym_name*/ StringAttr());
+        // Use the clock gate instead of input muxing.
+        Value currClockGate = notStalledClockGate;
+        for (auto hierClockGateEnable :
+             stageTerminator.getClockGatesForReg(regIdx)) {
+          // Create clock gates for any hierarchically nested clock gates.
+          currClockGate = builder.create<seq::ClockGateOp>(
+              loc, currClockGate, hierClockGateEnable, /*test_enable=*/Value());
+        }
+        dataReg = builder.create<seq::CompRegOp>(
+            stageTerminator->getLoc(), dataType, regIn, currClockGate, regName,
+            reset, /*resetValue*/ Value(), /*sym_name*/ StringAttr());
```

Comment:
I know this was there previously, but (at least on FPGAs) it's generally preferable not to reset data registers.

Reply:
Makes sense - i'll remove it. Can always add an option for ASIC (resets) in the future.

Reply:
I also don't see a reason to do it on ASICs, though I don't have any experience there.